### PR TITLE
Fix #26559: Edit Content Add Test for Utils Functions

### LINKED ISSUE #26559: Edit Content: Create Tests for Utils Functions
### Parent Issue

https://github.com/dotCMS/core/issues/25445

### Problem Statement

We have created a lot of utils in the Edit Content library and we haven't created any tests for them. We need to test them with random values and be sure that they don't break and work perfectly on edge cases.

### Steps to Reproduce

No STR

### Acceptance Criteria

All the utils functions should be covered in tests.

### dotCMS Version

master - latest

### Proposed Objective

Application Performance

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

- We should be sure that all utils functions have tests that covers all the cases.

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
🧪✨📝

This pull request improves the edit content library by enhancing the single selectable fields functionality and adding unit tests. It modifies the `functions.util.ts` file and adds a new file `functions.util.spec.ts` that contains the tests.

> _We are the masters of the code, we test and refactor_
> _We cast the values and the options, we make the content shine_
> _We document our functions well, we share our wisdom and our skill_
> _We are the metal coders, we rock the edit library_

### Walkthrough
*  Add unit tests for the functions in `functions.util.ts` ([link](https://github.com/dotCMS/core/pull/26569/files?diff=unified&w=0#diff-001fcca242290eaaace5b2b311d3fe695f710dc2c0dca7738785c21fef92e1adR1-R299))
*  Explain the purpose and usage of the `castSingleSelectableValue` function ([link](https://github.com/dotCMS/core/pull/26569/files?diff=unified&w=0#diff-99ac6732d97f8e73c67b5464dfb645a4ae4ec51a80005cf22d7f83a27dc43c74R13))
*  Improve the logic and readability of the `getSingleSelectableFieldOptions` function ([link](https://github.com/dotCMS/core/pull/26569/files?diff=unified&w=0#diff-99ac6732d97f8e73c67b5464dfb645a4ae4ec51a80005cf22d7f83a27dc43c74L35-R47))

### Screenshots
Screenshot 2023-10-30 at 2 03 50 PM

### CODE REVIEW

**Review by oidacra — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        });
+
+        describe('Boolean', () => {
+            it('should return true if value is true', () => {
+                expect(
+                    castSingleSelectableValue(
+                        'true',
+                        DotEditContentFieldSingleSelectableDataType.BOOL
+                    )
+                ).toBeTruthy();
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts:31]:
```suggestion
                ).toBe(true);
```

Comment (fmontes) [reply core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts:31]: This functions returns a boolean, so we can assert the booleat itself...

File: core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            });
+
+            it('should return false if value is false', () => {
+                expect(
+                    castSingleSelectableValue(
+                        'false',
+                        DotEditContentFieldSingleSelectableDataType.BOOL
+                    )
+                ).toBeFalsy();
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts:40]:
```suggestion
                ).toBe(false);
```

File: core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts (hunk 1 above)
Comment (fmontes) [reply core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts:31]: What happen if I send `asedfsafasfd` or a number? but the second value is BOOL?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/libs/edit-content/src/lib/utils/functions.util.spec.ts:31]: it should return `false` in that case, I'll a test for that

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]
